gateway/handler_error.go

```diff
@@ -87,6 +92,25 @@ type TemplateExecutor interface {
 	Execute(wr io.Writer, data interface{}) error
 }
 
+func (e *ErrorHandler) recordAccessLog(log *logrus.Logger, req *http.Request, resp *http.Response) {
```

Comment:
We could avoid repeating `recordAccessLog` on `ErrorHandler` and `SuccessHandler` by refactoring `recordAccessLog` to be a receiver method on `BaseMiddleware` which is embedded in `SuccessHandler` and `ErrorHandler`. 
```
func (t *BaseMiddleware) recordAccessLog(latency *analytics.Latency, log *logrus.Logger, req *http.Request, resp *http.Response) {
	// Don't print the full token, handle as obfuscated key or hashed key for security reasons
	hashKeys := t.Gw.GetConfig().HashKeys
	token := ctxGetAuthToken(req)

	if !hashKeys {
		token = t.Gw.obfuscateKey(token)
	} else {
		token = storage.HashKey(token, hashKeys)
	}

	accessLog := httputil.NewAccessLogRecord(t.Spec.APIID, token, t.Spec.OrgID)
	accessLog.WithLatency(latency)
	accessLog.WithRequest(req)
	accessLog.WithResponse(resp)

	log.WithFields(accessLog.Fields()).Info()
}
```

Reply:
fixed some api signature smells, and added an WithAuthToken to access log record;